parser_library/test/processing/ca_instr_test.cpp

```diff
@@ -27,341 +27,247 @@ using namespace hlasm_plugin::parser_library::semantics;
+    EXPECT_EQ(get_var_value<A_t>(a.hlasm_ctx(), "VAR3"), 0);
+}
 
-    auto it = ctx.ids().find("var");
-    auto it2 = ctx.ids().find("var2");
-    auto it3 = ctx.ids().find("var3");
+TEST(var_subs, big_arrays)
+{
+    std::string input = R"(
+    LCLC &LARR(100000000)
+    GBLC &GARR(100000000)
+)";
+    analyzer a(input);
+    a.analyze();
 
-    ASSERT_TRUE(ctx.get_var_sym(it));
-    ASSERT_TRUE(ctx.get_var_sym(it2));
-    ASSERT_TRUE(ctx.get_var_sym(it3));
+    EXPECT_EQ(get_var_vector<C_t>(a.hlasm_ctx(), "LARR")->size(), 0);
+    EXPECT_EQ(get_var_vector<C_t>(a.hlasm_ctx(), "GARR")->size(), 0);
 }
 
 TEST(var_subs, set_to_var)
 {
-    std::string input("&var seta 3");
+    std::string input("&VAR SETA 3");
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-
-    auto it = ctx.ids().find("var");
-
-    ASSERT_TRUE(ctx.get_var_sym(it));
-
-    diagnostic_op_consumer_container diags;
-    int tmp = get_var_sym_value(ctx, it, std::vector<int> {}, {}, diags).access_a();
-    EXPECT_EQ(tmp, 3);
+    EXPECT_EQ(get_var_value<A_t>(a.hlasm_ctx(), "VAR"), 3);
 }
 
 TEST(var_subs, set_to_var_idx)
 {
-    std::string input("&var(2) seta 3");
+    std::string input("&VAR(2) SETA 3");
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-    diagnostic_op_consumer_container diags;
-
-    auto it = ctx.ids().find("var");
-
-    ASSERT_TRUE(ctx.get_var_sym(it));
-    std::vector<A_t> subscript1;
-    subscript1.push_back(2);
-    int tmp = get_var_sym_value(ctx, it, std::move(subscript1), {}, diags).access_a();
-    EXPECT_EQ(tmp, 3);
+    std::unordered_map<size_t, A_t> expected = { { 1, 3 } };
+    EXPECT_THAT(*get_var_vector_map<A_t>(a.hlasm_ctx(), "VAR"), ::testing::ContainerEq(expected));
 }
 
 TEST(var_subs, set_to_var_idx_many)
 {
-    std::string input("&var(2) seta 3,4,5");
-    std::string input_s1("2");
-    std::string input_s2("3");
-    std::string input_s3("4");
-    analyzer s1(input_s1);
-    analyzer s2(input_s2);
-    analyzer s3(input_s3);
-
+    std::string input("&VAR(2) SETA 3,4,5");
     analyzer a(input);
     a.analyze();
 
-    auto& ctx = a.hlasm_ctx();
-    diagnostic_op_consumer_container diags;
-
-    auto it = ctx.ids().find("var");
-
-    ASSERT_TRUE(ctx.get_var_sym(it));
-
-    int tmp;
-    std::vector<A_t> subscript1;
-    subscript1.push_back(2);
-    tmp = get_var_sym_value(ctx, it, std::move(subscript1), {}, diags).access_a();
-    EXPECT_EQ(tmp, 3);
-    std::vector<A_t> subscript2;
-    subscript2.push_back(3);
-    tmp = get_var_sym_value(ctx, it, std::move(subscript2), {}, diags).access_a();
-    EXPECT_EQ(tmp, 4);
-    std::vector<A_t> subscript3;
-    subscript3.push_back(4);
-    tmp = get_var_sym_value(ctx, it, std::move(subscript3), {}, diags).access_a();
-    EXPECT_EQ(tmp, 5);
+    std::unordered_map<size_t, A_t> expected = { { 1, 3 }, { 2, 4 }, { 3, 5 } };
+    EXPECT_THAT(*get_var_vector_map<A_t>(a.hlasm_ctx(), "VAR"), ::testing::ContainerEq(expected));
```

Comment:
should work, right?

```suggestion
    EXPECT_EQ(get_var_vector_map<A_t>(a.hlasm_ctx(), "VAR"), expected);
```

Reply:
It does work. I wanted to uses `EXPECT_THAT` as it gives a nicer output message when something goes wrong. Nevertheless, I can change it to `EXPECT_EQ`

Reply:
My main problem is that the `*` on valueless optional is UB. But comparing it should be perfectly fine.